### ISSUE TITLE
Module::handleEvent needs access to it's properties.

### ISSUE THREAD

**Issue by Vassyli** [lotgd/core#66]
We should supply handleEvent as well with the module entity in order to access properties. This already is the case for onRegister/onUnregister. There are 2 possibilities:
- we provide the module in the context array
- we provide an additional parameter just as with onRegister/onUnregister. The function signature should be compatible as long as we add a parameter to the end of the function.

**Comment by austenmc** [comment]
Hmm. Tough one. Module is not the only class that implements the event listener pattern.

**Comment by Vassyli** [comment]
Yep. Though the issue is not about extending the interface (adding additional nullable parameters are allowed by php), it's more that EventManager now needs to tingle with the ModuleManager as well.

How about suscribing module events in a separate table and let ModuleManager suscribe to all events (using a pattern that matches everything) which then propagates the event to the modules?
